libcontainer/state_linux.go

```diff
@@ -35,6 +35,17 @@ type containerState interface {
 }
 
 func destroy(c *Container) error {
+	// Usually, when a container init is gone, all other processes in its
+	// cgroup are killed by the kernel. This is not the case for a shared
+	// PID namespace container, which may have some processes left after
+	// its init is killed or exited.
+	//
+	// As the container without init process running is considered stopped,
+	// and destroy is supposed to remove all the container resources, we need
+	// to kill those processes here.
+	if !c.config.Namespaces.IsPrivate(configs.NEWPID) {
+		_ = signalAllProcesses(c.cgroupManager, unix.SIGKILL)
+	}
 	err := c.cgroupManager.Destroy()
```

Comment:
We say we should not ignore the error from destroy, it means that we have a opportunity to destroy it again once we got an error, but if we ignore this cgroup destroy error, the state directory would be removed in next several lines, so the container disappears in runc, but leave some cgroup paths or some processes of this container in the host, we can’t delete them anymore from Runc.

Reply:
You are right, but I do not modify this code here. I surely can. 

Are you suggesting something like this?
```go
if err := c.cgroupManager.Destroy(); err != nil {
   return fmt.Errorf("can't remove container cgroup: %w", err)
}
```

Reply:
@lifubang added, PTAL

Reply:
> Are you suggesting something like this?

Yes, but I’m puzzled that why we ignored this error before? I will spend sometime to look git log to find out the reason if exists.

Reply:
> > Are you suggesting something like this?
> 
> Yes, but I’m puzzled that why we ignored this error before? I will spend sometime to look git log to find out the reason if exists.

Apparently you don't read commit messages 😄 I already looked and wrote about it.